tidy3d/components/simulation.py

```diff
@@ -4792,35 +4789,35 @@ def perturbed_mediums_copy(
         for carrier, data in zip(
             ["electron_density", "hole_density"], [electron_density, hole_density]
         ):
-            if isinstance(data, TetrahedralGridVoltageDataset) or isinstance(
-                data, TriangularGridVoltageDataset
-            ):
-                # make sure only one voltage is present
-                num_voltages = len(electron_density.values.voltage.data)
-                new_values = IndexedDataArray(
-                    np.array(data.values.data).flatten(), coords=dict(index=data.values.index.data)
-                )
-                if num_voltages > 1:
-                    raise ValueError(
-                        f"The value entered for '{carrier}' contains field values associated to "
-                        f"{num_voltages} voltage values. Please select one before calling this function. "
-                        "This can be done with 'electron_data.sel(voltage=1)'"
+            if isinstance(data, TriangularGridDataset) or isinstance(data, TetrahedralGridDataset):
+                if "voltage" in data.values.dims:
```

Comment:
for generality, we could probably switch this to 
```
if data._num_fields > 1:
```

Reply:
Not sure I like this since we're later checking for the additional fields. In this case `voltage`.